plugin-gradle/README.md

```diff
@@ -291,6 +293,142 @@ spotless {
+To use tsfmt, you first have to specify the files that you want it to apply to.
+Then you specify `tsfmt()`, and optionally how you want to apply it.
+
+By default, all typescript source sets will be formatted. To change this,
+set the `target` parameter as described in the [Custom rules](#custom) section.
+
+```gradle
+spotless {
+  typescript {
+    // using existing config files
+    tsfmt().tslintFile('/path/to/repo/tslint.json')
+  }
+}
+```
+Supported config file types are `tsconfigFile`, `tslintFile`, `vscodeFile` and `tsfmtFile`. They are corresponding to the respective
+[tsfmt-parameters](https://github.com/vvakame/typescript-formatter/blob/7764258ad42ac65071399840d1b8701868510ca7/lib/index.ts#L27L34).
+
+*Please note:*
+The auto-discovery of config files (up the file tree) will not work when using tsfmt within spotless,
+  hence you are required to provide resolvable file paths for config files.
+
+... or alternatively provide the configuration inline ...
+
+```gradle
+spotless {
+  typescript {
+    // custom file-set
+    target 'src/main/resources/**/*.ts'
+    // provide config inline
+    tsfmt().config(['indentSize': 1, 'convertTabsToSpaces': true])
+  }
+}
+```
+
+See [tsfmt's default config settings](https://github.com/vvakame/typescript-formatter/blob/7764258ad42ac65071399840d1b8701868510ca7/lib/utils.ts#L11L32) for what is available.
+
+... and it is also possible to apply `prettier()` instead of `tsfmt()` as formatter. For details see the section about [prettier](#typescript-prettier).
+
+### Prerequisite: tsfmt requires a working NodeJS version
+
+tsfmt is based on NodeJS, so to use it, a working NodeJS installation (especially npm) is required on the host running spotless.
+Spotless will try to auto-discover an npm installation. If that is not working for you, it is possible to directly configure the npm binary to use.
+
+```gradle
+spotless {
+  typescript {
+    tsfmt().npmExecutable('/usr/bin/npm').config(...)
+  }
+}
+```
+
+<a name="prettier"></a>
+
+## Applying [Prettier](https://prettier.io) to javascript | flow | typeScript | css | scss | less | jsx | graphQL | yaml | etc.
+
+Prettier is a formatter that can format [multiple file types](https://prettier.io/docs/en/language-support.html).
+
+To use prettier, you first have to specify the files that you want it to apply to.  Then you specify prettier, and how you want to apply it.
+
+```gradle
+spotless {
+  format 'styling', {
+    target '**/*.css', '**/*.scss'
+
+    // at least provide the parser to use
+    prettier().config(['parser': 'postcss'])
+
+    // or provide a typical filename
+    prettier().config(['filepath': 'style.scss'])
+  }
+}
+```
+
+Supported config options are documented on [prettier.io](https://prettier.io/docs/en/options.html).
+
+It is also possible to specify the config via file:
+
+```gradle
+spotless {
+  format 'styling', {
+    target '**/*.css', '**/*.scss'
+
+    // or provide both (config options will win over configFile options)
```

Comment:
`will win` -> `take precedence`? (Same for following comment.)

Reply:
thx